### DISCUSSION TITLE
urlopen error when trying to use pyarmor reg

### DISCUSSION THREAD

**Discussion by DanPSA** [dashingsoft/pyarmor#1650] [Q&A]
I have purchased a pyarmor pro license, received the pyarmor-regcode-XXXX.txt file, and when trying to use the pyarmor reg command with that file it results in an ERROR.

Here is the output from the command:

>pyarmor reg -p "<TOOLNAME>" pyarmor-regcode-XXXX.txt
INFO     Python 3.11.1
INFO     Pyarmor 8.4.6 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     prepare "pyarmor-regcode-5496.txt"
INFO     query key file from server
ERROR    <urlopen error _ssl.c:975: The handshake operation timed out>

I have entered an exception in my firewall software for pyarmor and still have the same issue.

When I attempt to ping api.dashingsoft.com I get the following:

>ping api.dashingsoft.com
Pinging api.dashingsoft.com [119.23.58.77] with 32 bytes of data:
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Ping statistics for 119.23.58.77:
    Packets: Sent = 4, Received = 0, Lost = 4 (100% loss),

I'm at a loss as to what to do next.  Does this indicate an issue on dashingsoft's end?

**Comment by jondy** [comment]
api.dashingsoft.com will block ping request, I don't think it's pyarmor server issue, it works fine and serve others requests well.
Please check your firewire settings, also check pyarmor documentation

**Comment by DanPSA** [comment]
It appears the issue was resolved by first running CMD in windows as administrator.  The pyarmor reg command now works as expected.

I realize that I accidently left off part of the error message in what I posted above.  I no longer have that text, but it referenced a urlopen error.
